scene/gui/text_edit.cpp

```diff
@@ -4504,6 +4560,68 @@ int TextEdit::get_caret_count() const {
 	return carets.size();
 }
 
+void TextEdit::insert_caret_at_carets(bool p_below) {
+	for (const int &caret_index : caret_index_edit_order) {
+		const int caret_line = get_caret_line(caret_index);
+		const int caret_column = get_caret_column(caret_index);
+
+		// The last fit x will be cleared if the caret has a selection,
+		// but if it does not have a selection the last fit x will be
+		// transferred to the new caret
+		int caret_from_column = 0, caret_to_column = 0, caret_last_fit_x;
+		if (has_selection(caret_index)) {
+			// If the selection goes over multiple lines, deselect it.
+			if (get_selection_from_line(caret_index) != get_selection_to_line(caret_index)) {
+				deselect(caret_index);
+				caret_last_fit_x = carets[caret_index].last_fit_x;
+			} else {
+				caret_from_column = get_selection_from_column(caret_index);
+				caret_to_column = get_selection_to_column(caret_index);
+				caret_last_fit_x = -1;
+				carets.write[caret_index].last_fit_x = _get_column_x_offset_for_line(caret_column, caret_line, caret_column);
+			}
+		} else {
+			caret_last_fit_x = carets[caret_index].last_fit_x;
+		}
+
+		// Get the line and column of the new caret as if you would move the caret by pressing the arrow keys
+		int new_caret_line, new_caret_column, new_caret_from_column = 0, new_caret_to_column = 0;
+		_get_new_caret_line_column(caret_line, get_caret_wrap_index(caret_index), caret_column, p_below, new_caret_line, new_caret_column, caret_last_fit_x);
+
+		// If the caret does have a selection calculate the new from and to columns
+		if (caret_from_column != caret_to_column) {
+			// We only need to calculate the selection columns if the column of the caret changed
+			if (caret_column != new_caret_column) {
+				int _; // unused placeholder for p_new_line
+				_get_new_caret_line_column(caret_line, get_caret_wrap_index(caret_index), caret_from_column, p_below, _, new_caret_from_column);
+				_get_new_caret_line_column(caret_line, get_caret_wrap_index(caret_index), caret_to_column, p_below, _, new_caret_to_column);
+			} else {
+				new_caret_from_column = caret_from_column;
+				new_caret_to_column = caret_to_column;
+			}
+		}
+
+		// Add the new caret
+		const int new_caret_index = add_caret(new_caret_line, new_caret_column);
+
+		if (new_caret_index != -1) {
+			// Also add the selection if there should be one
+			if (new_caret_from_column != new_caret_to_column) {
+				select(new_caret_line, new_caret_from_column, new_caret_line, new_caret_to_column, new_caret_index);
+				// Necessary to properly modify the selection after adding the new caret
+				carets.write[new_caret_index].selection.selecting_line = new_caret_line;
+				carets.write[new_caret_index].selection.selecting_column = new_caret_column == new_caret_from_column ? new_caret_to_column : new_caret_from_column;
+			} else {
+				// Copy the last fit x over
+				carets.write[new_caret_index].last_fit_x = carets[caret_index].last_fit_x;
+			}
+		}
```

Comment:
`Nit:` can invert here:
```suggestion
		if (new_caret_index == -1) {
		    continue;
		}
		// Also add the selection if there should be one
		if (new_caret_from_column != new_caret_to_column) {
			select(new_caret_line, new_caret_from_column, new_caret_line, new_caret_to_column, new_caret_index);
			// Necessary to properly modify the selection after adding the new caret
			carets.write[new_caret_index].selection.selecting_line = new_caret_line;
			carets.write[new_caret_index].selection.selecting_column = new_caret_column == new_caret_from_column ? new_caret_to_column : new_caret_from_column;
			continue;
		} 

		// Copy the last fit x over
		carets.write[new_caret_index].last_fit_x = carets[caret_index].last_fit_x;
```